pkg/odo/cli/service/create.go

```diff
@@ -447,56 +506,46 @@ func NewCmdServiceCreate(name, fullName string) *cobra.Command {
 	return serviceCreateCmd
 }
 
-// Parses group and version values from the alm-example
-func groupVersionALMExample(example map[string]interface{}) (group, version string) {
-	apiVersion := example["apiVersion"].(string)
-	// use SplitN so that if apiVersion field's value is something like
-	// etcd.coreos.com/v1/beta1 then group's value ends up being etcd.cores.com
-	// and version ends up being v1/beta1
-	gv := strings.SplitN(apiVersion, "/", 2)
-
-	group, version = gv[0], gv[1]
-	return
-}
-
-func resourceFromCSV(csv olmv1alpha1.ClusterServiceVersion, crdName string) (resource string) {
-	for _, crd := range csv.Spec.CustomResourceDefinitions.Owned {
-		if crd.Kind == crdName {
-			resource = strings.Split(crd.Name, ".")[0]
-			return
-		}
+// validateMetadataInCRD validates if the CRD has metadata.name field and returns an error
+func (d *DynamicCRD) validateMetadataInCRD() error {
+	metadata, ok := d.OriginalCRD["metadata"].(map[string]interface{})
+	if !ok {
+		// this condition is satisfied if there's no metadata at all in the provided CRD
+		return fmt.Errorf("Couldn't find \"metadata\" in the yaml. Need metadata start the service")
 	}
-	return
-}
 
-func getAlmExample(almExamples []map[string]interface{}, crd, operator string) (map[string]interface{}, error) {
-	for _, example := range almExamples {
-		if example["kind"].(string) == crd {
-			return example, nil
-		}
+	if _, ok := metadata["name"].(string); ok {
+		// found the metadata.name; no error
+		return nil
 	}
-	return nil, errors.Errorf("Could not find example yaml definition for %q service in %q operator's definition.\nPlease provide a file containing yaml specification to start the service from operator\n", crd, operator)
+	return fmt.Errorf("Couldn't find metadata.name in the yaml. Provide a name for the service")
 }
 
-func doesCRExist(kind string, csvs *olmv1alpha1.ClusterServiceVersionList) (olmv1alpha1.ClusterServiceVersion, error) {
-	for _, csv := range csvs.Items {
-		for _, operatorCR := range csv.Spec.CustomResourceDefinitions.Owned {
-			if kind == operatorCR.Kind {
-				return csv, nil
-			}
+// setServiceName modifies the CRD to contain user provided name on the CLI
+// instead of using the default one in almExample
+func (d *DynamicCRD) setServiceName(name string) {
+	metaMap := d.OriginalCRD["metadata"].(map[string]interface{})
+
+	for k := range metaMap {
+		if k == "name" {
```

Comment:
Something more descriptive than `k`

Reply:
It actually helps to keep short names at places like these. IMO, it's not always helpful/productive to use long names. `for` loops are one such place. And this one in particular doesn't make sense to me because we're iterating over a `map` which is evident from the name `metaMap`. So it's quite straightforward that if you iterate as `for k, v := range metaMap`, `k` is key and `v` is value, whereas if you iterate `for k := range metaMap`, `k` is the key. I do not think we need a more descriptive name here.

Reply:
OK